app/models/mute.rb

```diff
@@ -22,9 +24,15 @@ class Mute < ApplicationRecord
 
   after_commit :remove_blocking_cache
 
+  before_destroy :delete_unmute_job
+
   private
 
   def remove_blocking_cache
     Rails.cache.delete("exclude_account_ids_for:#{account_id}")
   end
+
+  def delete_unmute_job
+    Sidekiq::ScheduledSet.new.find_job(unmute_jid)&.delete if unmute_jid
```

Comment:
I wonder if it wouldn't be best to just let the job execute and die rather than remove it from the queue, as going through the queue to remove it might be costly.

Reply:
I think the process of deleting the job is necessary.

In case of you unmute and mute the same account again manually (with different duration value), the process of unmuting may be mistakenly performed if the job is remained.

Reply:
Ah, then you could pass a mute id instead, which also seems more natural… but on the other hand, the `UnmuteService` does take a pair of account `id`s and not a mute `id`, so you'd have to get the `id`s from the mute object… which is fine, just a bit awkward

Reply:
I changed to pass the ```mute_id``` to the worker, instead of ```account_id``` and ```target_account_id```. Job deletion is now unnecessary.